config/m4/compiler.m4

```diff
@@ -570,7 +570,7 @@ ADD_COMPILER_FLAGS_IF_SUPPORTED([[-Wno-missing-field-initializers],
 #
 # Intel Compiler specifics
 #
-if ! $CC -V | grep -q Intel.*Compiler; then
+if ! $CC --version | grep -q Intel; then
```

Comment:
i would convert | to |& to work on compilers that still do not support --version flag

Reply:
added &

Reply:
@yosefe please re-check changed to:
`if $CC --version 2>&1 | grep -q Intel; then`